src/ocm/components/clusterConfiguration/networkConfiguration/SubnetsDropdown.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { DropdownItem, DropdownToggle, Dropdown } from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons';
+import { useField } from 'formik';
+import { getFieldId, HostSubnet, NO_SUBNET_SET } from '../../../../common';
+
+type SubnetsDropdownProps = {
+  name: string;
+  machineSubnets: HostSubnet[];
+  isDisabled: boolean;
+};
+
+const toFormSelectOptions = (subnets: HostSubnet[]) => {
+  return subnets.map((hn, index) => ({
+    label: `${hn.humanized}${hn.isValid ? '' : ' (Invalid network)'}`,
+    value: hn.subnet,
+    isDisabled: false,
+    id: `form-input-hostSubnet-field-option-${index}`,
+  }));
+};
+
+const makeNoSubnetSelectedOption = (availableSubnets: number) => ({
+  label: `Please select a subnet. (${availableSubnets} available)`,
+  value: NO_SUBNET_SET,
+  isDisabled: true,
+  id: 'form-input-hostSubnet-field-option-no-subnet-selected',
+});
+
+const noSubnetAvailableOption = {
+  label: 'No subnets are currently available',
+  value: NO_SUBNET_SET,
+  isDisabled: true,
+  id: 'form-input-hostSubnet-field-option-no-subnet-available',
+};
+
+export const SubnetsDropdown = ({ name, machineSubnets, isDisabled }: SubnetsDropdownProps) => {
+  const [field, , { setValue }] = useField(name);
+  const [isOpen, setOpen] = React.useState(false);
+  const [current, setCurrent] = React.useState('');
+  const fieldId = getFieldId(name, 'input');
+
+  const buildOptions = React.useMemo(
```

Comment:
The `memo` should be based off `machineSubnets` and return the memo items,
otherwise I think we are losing the performance benefit of the memo.

```
const items = React.useMemo(..., [machineSubnets])
```